File: src/main/java/com/sem/keeper/model/SemUserDetail.java
package com.sem.keeper.model;

import com.sem.keeper.entity.UserEntity;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.User;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public class SemUserDetail extends User {

    public SemUserDetail(UserEntity user){
        super(user.getEmail(),user.getPassword(),
                true,true,true,true,
                getAuthorities(user.getRoles()));
    }

    public SemUserDetail(String username, String password, Collection<? extends GrantedAuthority> authorities) {
        super(username, password, authorities);
    }

    public SemUserDetail(String username, String password, boolean enabled, boolean accountNonExpired, boolean credentialsNonExpired, boolean accountNonLocked, Collection<? extends GrantedAuthority> authorities) {
        super(username, password, enabled, accountNonExpired, credentialsNonExpired, accountNonLocked, authorities);
    }

    private static List<GrantedAuthority> getAuthorities (List<String> roles) {
        List<GrantedAuthority> authorities = new ArrayList<>();
        for (String role : roles) {
            authorities.add(new SimpleGrantedAuthority(role));
        }
        return authorities;
    }
}

File: src/main/java/com/sem/keeper/model/SemUserDetail.kt
package com.sem.keeper.model

import com.sem.keeper.entity.UserEntity
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.User
import java.util.ArrayList



class SemUserDetail(user: UserEntity) :
    User(
        user.email, user.password,
        true, true, true, true,
        getAuthorities(user.roles!!)
    ) {
    companion object {
        private fun getAuthorities(roles: List<String>): List<GrantedAuthority> {
            val authorities: MutableList<GrantedAuthority> = ArrayList()
            for (role in roles) {
                authorities.add(SimpleGrantedAuthority(role))
            }
            return authorities
        }
    }
}
